src/test/java/game/map/MapValidatorTest.java

```diff
@@ -0,0 +1,256 @@
+package game.map;
+
+import static game.map.MapLoader.loadMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+import game.map.MapValidator;
+import game.pojo.Continent;
+import game.pojo.Country;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class MapValidatorTest {
```

Comment:
Before reviewing tests, I would like to confirm my understanding,

what does connectivity mean actually?

-> from any country in the map we should be able to reach any other country in the map
-> from any country in a continent we should be able to reach any other country in the continent.

Is this understanding correct?

Reply:
I would phrase the definition slightly differently when it comes to continent.
For continents, you only consider the subgraph induced by the continent (i.e. only the countries that belong to the continent). Then, the continent being connected means that for any 2 countries in the subgraph, you can build a path (that consists only of other countries in the subgraph) that connects these 2 countries.

With the way you're defining connectivity, if the first statement is true, then the second one is automatically true

Reply:
With the way you're defining connectivity, if the first statement is true, then the second one is automatically true
    - Yes, I thought we don't to check the connectivity of a continent separately. But I am wrong.

Thanks for explaining that when checking the connectivity of a continent, we have to pass only through the countries of that continent.

Reply:
Tests look good too. Please feel free to merge if you are okay with this.

Reply:
I've changed to class to accomodate with what you said about the ```d_mapToValidate``` atribute, now the map has to be given as argument to every method